### ISSUE TITLE
MissingValueException for `javaCompiler` property with new `JdkImageWorkaround`

### ISSUE THREAD

**Issue by ZacSweers** [gradle/android-cache-fix-gradle-plugin#307]
After trying `2.5.4`, we see this issue

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> Could not create task ':app:compileInternalDebugJavaWithJavac'.
   > Cannot query the value of task ':app:compileInternalDebugJavaWithJavac' property 'javaCompiler' because it has no value available.

// skipping stuff for brevity

Caused by: org.gradle.api.internal.provider.MissingValueException: Cannot query the value of task ':app:compileInternalDebugJavaWithJavac' property 'javaCompiler' because it has no value available.
        at org.gradle.api.internal.provider.AbstractMinimalProvider.get(AbstractMinimalProvider.java:85)
        at org.gradle.api.provider.Provider$get.call(Unknown Source)
        at org.gradle.android.workarounds.JdkImageWorkaround$_replaceCommandLineProvider_closure7$_closure12.doCall(JdkImageWorkaround.groovy:99)

```

**Comment by runningcode** [comment]
@ZacSweers thanks for filling. Do you know what version of Gradle, AGP and Kotlin this is happening on?

**Comment by runningcode** [comment]
As an FYI, you can disable the workaround with the system property `org.gradle.android.cache-fix.JdkImageWorkaround.enabled=false`.

**Comment by ZacSweers** [comment]
Gradle 7.5-rc-2, AGP 7.2.1, KGP 1.7.0

**Comment by eygraber** [comment]
Same happening for me on Gradle 7.4.2, AGP 7.2.1, and KGP 1.6.21

**Comment by ghale** [comment]
This should be fixed in [version 2.5.5](https://plugins.gradle.org/plugin/org.gradle.android.cache-fix).
